src/core/contract_address/starknet_contract_address.rs

```diff
@@ -30,12 +30,15 @@ fn get_contract_entry_points(
     entry_point_type: &EntryPointType,
 ) -> Result<Vec<ContractEntryPoint>, ContractAddressError> {
     let program_length = contract_class.program.data.len();
+
     let entry_points = contract_class
         .entry_points_by_type
         .get(entry_point_type)
         .ok_or(ContractAddressError::NoneExistingEntryPointType)?;
+
     for entry_point in entry_points {
-        if (Felt::from(0) <= entry_point.offset) && (entry_point.offset < program_length.into()) {
+        if !((Felt::from(0) <= entry_point.offset) && (entry_point.offset < program_length.into()))
```

Comment:
Maybe I'm mistaken, but aren't `Felt`s unsigned? If that is so, then the first conditional will always evaluate to `true`.
For the other condition, that should work but I assume that it's more expensive to convert an integer to a `Felt` than the other way around, and since any `Felt` that doesn't fit an `usize` will never be smaller than `program_length` you could try converting it to an `usize` and checking if it's smaller, returning `false` when the conversion fails.

Reply:
you are right about the first condition, i remove it. For the other i don't think it is going to impact that much, something that i can do is to take the conversion out of the loop and when we bench mark it, if we detect that it is a bottleneck, we change it.

Reply:
I guess it depends on whether the `Felt` allocates memory or not.